cypress-io/cypress
# fix: fixed issue with CT + electron + run mode not exiting properly

- Closes #25568

### Additional details

Removed unneeded code that was causing component tests to hang in `electron` when running multiple specs in `run` mode.

Prior to d1dc1258eafa6ff047abc913ebb930984e02a38d, after running the first spec in electron, the browser instance is [cleared](https://github.com/cypress-io/cypress/blob/8d3a6eed9293c72109bd10a77b2a79e922962705/packages/server/lib/browsers/index.ts#L173) and never set again even after a new spec is started. When `openProject.closeBrowser()` was called, the instance was [null](https://github.com/cypress-io/cypress/blob/8d3a6eed9293c72109bd10a77b2a79e922962705/packages/server/lib/browsers/index.ts#L21) and the promise resolved.

After d1dc1258eafa6ff047abc913ebb930984e02a38d, after running the first spec in electron, the browser instance is [cleared](https://github.com/cypress-io/cypress/blob/8d3a6eed9293c72109bd10a77b2a79e922962705/packages/server/lib/browsers/index.ts#L173) but then it is set to the [new browser instance](https://github.com/cypress-io/cypress/blob/8d3a6eed9293c72109bd10a77b2a79e922962705/packages/server/lib/browsers/index.ts#L130) when the new spec is started. The instance is not cleared again after that. Thus, when `openProject.closeBrowser()` was called, the instance was [not null](https://github.com/cypress-io/cypress/blob/8d3a6eed9293c72109bd10a77b2a79e922962705/packages/server/lib/browsers/index.ts#L21) and the promise never resolved causing the hang.

### Steps to test

* Run multiple component specs in `run` mode in `electron` and verify the process is correctly exited.

### How has the user experience changed?



### PR Tasks


- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
